Makefile

```
@@ -1,162 +1,158 @@
-	cp static/dist/preview.js ${DCL_PROJECT}/node_modules/decentraland-ecs/artifacts/
+static/systems/decentraland-ui.scene.js: $(SCENE_SYSTEM) packages/ui/tsconfig.json packages/ui/decentraland-ui.scene.ts
+	@$(COMPILER) targets/engine/internal-scenes.json
 
-publish:
-	@echo "$(GREEN)=================== getting release ready ==================$(RESET)"
-	$(NODE) ./scripts/prepareDist.js
+packages/decentraland-ecs/dist/src/index.js: $(DECENTRALAND_ECS_SOURCES) $(ECS_CONFIG_DEPENDENCIES)
+	@$(COMPILER) targets/engine/ecs.json
 
-	@echo "$(GREEN)===================== decentraland-ecs =====================$(RESET)"
-	@cd $(PWD)/packages/decentraland-ecs; node $(PWD)/scripts/npmPublish.js
+packages/decentraland-ecs/types/dcl/index.d.ts: $(COMPILED_SUPPORT_JS_FILES) $(DECENTRALAND_ECS_COMPILED_FILES) $(ECS_CONFIG_DEPENDENCIES)
+	@cd $(PWD)/packages/decentraland-ecs; $(PWD)/node_modules/.bin/api-extractor run --typescript-compiler-folder "$(PWD)/node_modules/typescript" --local
+	@node ./scripts/buildEcsTypes.js
+	@npx prettier --write 'packages/decentraland-ecs/types/dcl/index.d.ts'
+	@touch $(DECENTRALAND_ECS_TYPEDEF_FILE)
+
+docs: $(DECENTRALAND_ECS_TYPEDEF_FILE) ## Generate `decentraland-ecs` documentation using @microsoft/api-documenter
+	@cd $(PWD)/packages/decentraland-ecs; $(PWD)/node_modules/.bin/api-documenter markdown -i types/dcl -o docs
+	@echo "\nAPI Documentation was generated in the packages/decentraland-ecs/docs folder"
+
+build-essentials: $(COMPILED_SUPPORT_JS_FILES) $(DECENTRALAND_ECS_TYPEDEF_FILE) $(BUILD_ECS) $(SCENE_SYSTEM) $(INTERNAL_SCENES) $(DECENTRALAND_LOADER) ## Build the basic required files for the explorer
+
+# Hellmap scenes
+
+HELLMAP_SOURCE_FILES := $(wildcard public/hell-map/*/game.ts)
+HELLMAP_GAMEJS_FILES := $(subst .ts,.js,$(HELLMAP_SOURCE_FILES))
+
+public/hell-map/%/game.js: $(SCENE_SYSTEM) public/hell-map/%/game.ts
+	@$(COMPILER) targets/scenes/hell-map.json
+
+# Test scenes
+
+TEST_SCENES_SOURCE_FILES := $(wildcard public/test-scenes/*/game.ts)
+TEST_SCENES_GAMEJS_FILES := $(subst .ts,.js,$(TEST_SCENES_SOURCE_FILES))
+
+public/test-scenes/%/game.js: $(SCENE_SYSTEM) public/test-scenes/%/game.ts
+	@$(COMPILER) targets/scenes/test-scenes.json
+
+TEST_ECS_SCENE_SOURCES := $(wildcard public/ecs-scenes/*/game.ts)
+TEST_ECS_SCENE_GAMEJS_FILES := $(subst .ts,.js,$(TEST_ECS_SCENE_SOURCES))
+
+# ECS scenes
 
-	@echo "$(GREEN)======================== build-ecs =========================$(RESET)"
+public/ecs-scenes/%/game.js: $(SCENE_SYSTEM) public/ecs-scenes/%/game.ts
+	@node scripts/buildECSprojects.js
+
+# All scenes together
+
+build-test-scenes: $(TEST_SCENES_GAMEJS_FILES) $(HELLMAP_GAMEJS_FILES) $(TEST_ECS_SCENE_GAMEJS_FILES) ## Build the test scenes
+
+# Entry points
+
+static/dist/%.js: build-essentials packages/entryPoints/%.ts
+	@$(COMPILER) $<
+	
+# Release
+
+DIST_ENTRYPOINTS := "static/dist/editor.js static/dist/unity.js static/dist/preview"
```

Removing quotes as it makes `make` think that the filename is `"static/dist/editor.js` and won't find the target.
```suggestion
DIST_ENTRYPOINTS := static/dist/editor.js static/dist/unity.js static/dist/preview
```